master/buildbot/changes/gitpoller.py

```diff
@@ -159,8 +170,16 @@ def poll(self):
             '+%s:%s' % (self._removeHeads(branch), self._trackerBranch(branch))
             for branch in branches
         ]
-        yield self._dovccmd('fetch',
-                            [self.repourl] + refspecs, path=self.workdir)
+
+        while True:
+            try:
+                yield self._dovccmd('fetch', [self.repourl] + refspecs,
+                                    path=self.workdir)
+            except GitError, e:
+                log.msg(e)
+                time.sleep(self.pollInterval)
```

Comment:
Ah.. I hadn't thought that this might be a problem. I kept thinking that since we're using `@defer.inlineCallbacks` so everything can work as in the blocking case :smile: I'll need to read twisted docs some more, I guess.